core/server/services/members/middleware.js

```diff
@@ -182,6 +182,17 @@ const createSessionFromMagicLink = async function (req, res, next) {
             }
         }
 
+        if (action === 'signin') {
+            const referrer = req.query.referrer;
+            const siteUrl = urlUtils.getSiteUrl();
+
+            if (referrer && referrer.startsWith(siteUrl)) {
```

Comment:
I think this check could be better maybe? Need to investigate edgecases

Reply:
The default is `strict-origin-when-cross-origin` which works fine with this - we can choose to explicitly set one though!